ssht/plugins.py

```diff
@@ -128,3 +129,22 @@ def _load_data(self):
 
                 for row in c:
                     self._hosts.append(Host.factory(row))
+
+class APIParser(Parser):
+    def __init__(self, *args, **kwargs):
+        super(APIParser, self).__init__(*args, **kwargs)
+        self._files = self.get_files('.api')
+        self._load_data()
+
+    def _load_data(self):
+        for file_ in self._files:
+            path = os.path.join(self._path, file_)
+            logging.debug('Parsing "{0}"'.format(path))
+            with open(path, 'r') as fh:
+                d = json.loads(fh.read())
+                if 'headers' not in d['config']:
+                    d['config']['headers'] = {}
+                req = requests.get(d['config']['url'],
+                        headers=d['config']['headers'])
+                for host in req.json()['hosts']:
```

Comment:
Deze code gaat ervanuit dat:

- `json.loads()` niet failed
- dat `req` een valid response object teruggeeft (2XX status en een json response is)

`req` kun je beter `resp` noemen, want dat het is het. Vervolgens `resp.raise_for_status()` om alle niet 2XX status codes te filteren (middels een exception), daarna check je of het response idd json is en daarna check je of 'host' een valid key is van deze response dict en deze een list is.

Dan kun je erdoor loopen.